src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs

```diff
@@ -0,0 +1,132 @@
+        /// </summary>
+        [Output]
+        public string[] OutputSources { get; set; }
+
+        [Output]
+        public string OutputPackagesPath { get; set; }
+
+        [Output]
+        public string[] OutputFallbackFolders { get; set; }
+
+        [Output]
+        public string[] ConfigFilePaths { get; set; }
+
+        private static Lazy<IMachineWideSettings> _machineWideSettings = new Lazy<IMachineWideSettings>(() => new XPlatMachineWideSetting());
+
+        public ISettings GetSettings(string projectDirectory)
+        {
+            if (string.IsNullOrEmpty(RestoreConfigFile))
+            {
+                return Settings.LoadDefaultSettings(projectDirectory,
+                    configFileName: null,
+                    machineWideSettings: _machineWideSettings.Value);
+            }
+            else
+            {
+                var configFileFullPath = Path.GetFullPath(RestoreConfigFile);
+                var directory = Path.GetDirectoryName(configFileFullPath);
+                var configFileName = Path.GetFileName(configFileFullPath);
+                return Settings.LoadSpecificSettings(projectDirectory,
+                    configFileName: configFileName);
+            }
+        }
+
+        public override bool Execute()
+        {
+            // log inputs
+            var log = new MSBuildLogger(Log);
+            log.LogDebug($"(in) ProjectUniqueName '{ProjectUniqueName}'");
+            if (RestoreSources != null)
+            {
+                log.LogDebug($"(in) RestoreSources '{string.Join(";", RestoreSources.Select(p => p))}'");
+            }
+            if (RestorePackagesPath != null)
+            {
+                log.LogDebug($"(in) RestorePackagesPath '{RestorePackagesPath}'");
+            }
+            if (RestoreFallbackFolders != null)
+            {
+                log.LogDebug($"(in) RestoreFallbackFolders '{string.Join(";", RestoreFallbackFolders.Select(p => p))}'");
+            }
+            if (RestoreConfigFile != null)
+            {
+                log.LogDebug($"(in) RestoreConfigFile '{RestoreConfigFile}'");
+            }
+            var settings = GetSettings(Path.GetDirectoryName(ProjectUniqueName));
+
+            if (string.IsNullOrEmpty(RestorePackagesPath))
+            {
+                OutputPackagesPath = SettingsUtility.GetGlobalPackagesFolder(settings);
+            }
+            else
+            {
+                OutputPackagesPath = RestorePackagesPath;
+            }
+
+            if (RestoreSources == null)
+            {
+                var packageSourceProvider = new PackageSourceProvider(settings);
+                var packageSourcesFromProvider = packageSourceProvider.LoadPackageSources();
+                OutputSources = packageSourcesFromProvider.Select(e => e.Source).ToArray();
+            }
+            else
+            {
+                OutputSources = RestoreSources;
+            }
+
+            if (RestoreFallbackFolders == null)
```

Comment:
I think a 3rd option be useful. Currently you have:
empty -> settings
non-empty -> use value

It should also have:
'false' -> ignore settings and return an empty set

This would give projects the option to ignore nuget.config for certain things.  MSBuildStringUtility currently has an IsTrue helper, it could have an IsFalse also.